packages/admin/admin-stories/src/docs/components/DataGrid/DataGrid.stories.mdx

```diff
@@ -0,0 +1,108 @@
+
+#### Arguments
+
+| Name              | Type   | Description                                                                      |
+| :---------------- | :----- | :------------------------------------------------------------------------------- |
+| pageSize          | number | Number of items per page, defaults to 20.                                        |
+| queryParamsPrefix | string | Prefix used for query parameters, useful when multiple DataGrid are on one page. |
+
+#### Example
+
+<Canvas>
+    <Story id="stories-components-datagrid--usedatagridremote" />
+</Canvas>
+
+## usePersistentColumnState
+
+Returns props for DataGrid that persists all column states into localStorage, DataGrid will have a controlled state for those.
+
+Supported states:
+
+-   columnVisibility
+-   pinnedColumns (MUI X Pro feature)
+-   columnDimensions (MUI X Pro feature)
+-   columnOrder (MUI X Pro feature)
+
+#### Arguments
+
+| Name       | Type   | Description                                  |
+| :--------- | :----- | :------------------------------------------- |
+| stateKey\* | string | Unique String used as prefix in localStorage |
+
+#### Example
+
+<Canvas>
+    <Story id="stories-components-datagrid--usepersistentcolumnstate" />
+</Canvas>
+
+## useBufferedRowCount
+
+Small hook that can be used to prevent rowCount from being undefined during the loading.
+
+```
+const rowCount = useBufferedRowCount(data?.products.totalCount);
+```
+
+## GridFilterButton
+
+Small Component that can be placed in the Toolbar taht will show the Filter. Must be used as DataGrid child as shown in the story below.
+
+<Canvas>
+    <Story id="stories-components-datagrid--gridfilterbutton" />
+</Canvas>
+
+## CrudContextMenu
+
+Component that can be rendered as ContextMenu in a row that has some basic CRUD features:
+
+-   `url: string`: Shows a "Copy Url" Menu Item that copies the given Url into the clipboard
+-   `copyData: () => Promise<CopyData> | CopyData;`: Shows a "Copy" Menu Item, should return JSON representation of the row
+-   `onPaste: (options: { input: CopyData; client: ApolloClient<object> }) => Promise<void>;`: Shows as "Paste" Menu Item, implementation should insert a new row based on the passed `input` JSON. Attention: `input` is not validated.
+-   `onDelete: (options: { client: ApolloClient<object> }) => Promise<void>;`: Shows a "Delete" Menu Item, implementation should delete the row.
+-   `refetchQueries: RefetchQueriesOptions<any, unknown>["include"];`: called on apollo after executing delete or paste
+
+<Canvas>
+    <Story id="stories-components-datagrid--crudcontextmenu" />
+</Canvas>
+
+## DataGrid to Comet Graphl API Converter
+
+Comet Admin comes with two helper that convert MUI DataGrid state to standard Comet GraphQL API variables.
+
+-   `muiGridFilterToGql`
+-   `muiGridSortToGql`
+
+Typical Usage Example:
+
+```
+const dataGridProps = { ...useDataGridRemote(), ...usePersistentColumnState("ProductsGrid") };
```

Comment:
does this work with this syntax or is that just pseudo code? I thought hooks must be called inside a function and i think this is abusing this.

```suggestion
const dataGrid = useDataGridRemote(),
const persistentColumnState =usePersistentColumnState("ProductsGrid")
const dataGridProps = { ...dataGrid, ...persistentColumnState };
```

Reply:
This is perfectly working code and I see no reason for introducing 2 more variables. It also doesn't violate any hook rules or so.

Reply:
ok, have never seen that a hook is used like this, I just wondered if this is working and in constraint with hook rules.

if it's working - it is ok for me - even when I would not write it that way. :-)